packages/datatrak-web/src/views/AccountSettingsPage/RequestCountryAccessSection/RequestCountryAccessForm.tsx

```diff
@@ -0,0 +1,244 @@
+   * HACK: Force a fresh render of the country checklist. Use when a MUI checkbox appears checked,
+   * but is actually unchecked (or vice versa).
+   */
+  const rerenderCountryChecklist = () => setCountryChecklistKey(Date.now());
+
+  const sizeClassIsMdOrLarger = useMediaQuery(theme.breakpoints.up('sm'));
+
+  const formContext = useForm<RequestCountryAccessFormFields>({
+    defaultValues: {
+      countryIds: [],
+      reasonForAccess: '',
+    } as RequestCountryAccessFormFields,
+    mode: 'onChange',
+  });
+  const {
+    formState: { isSubmitting, isValidating, isValid },
+    handleSubmit,
+    register,
+    reset,
+  } = formContext;
+
+  const applicableCountries =
+    countries?.filter((country: TupaiaWebCountryAccessListRequest.CountryAccess) =>
+      project?.names?.includes(country.name),
+    ) ?? [];
+
+  const formIsNotSubmissible =
+    !project || isValidating || !isValid || isSubmitting || accessListIsLoading || requestIsLoading;
+
+  function onSubmit(formData: RequestCountryAccessFormFields) {
+    requestCountryAccess({
+      entityIds: formData.countryIds,
+      message: formData.reasonForAccess,
+      projectCode: projectCode,
+    });
+  }
+
+  return (
+    <StyledForm formContext={formContext} onSubmit={handleSubmit(onSubmit)}>
+      <StyledFieldset>
+        <CountryChecklistWrapper>
+          <StyledFormLabel>Select countries</StyledFormLabel>
+          <CountryChecklist
+            disabled={!project || isSubmitting || requestIsLoading}
+            key={countryChecklistKey}
+          >
+            {applicableCountries.map(({ id, name, hasAccess, accessRequests }) => {
+              const hasRequestedAccess = accessRequests.includes(ensure(projectCode));
+              const getTooltip = () => {
+                if (hasAccess) return 'You already have access';
+                if (hasRequestedAccess) return 'Approval in progress';
+              };
+
+              return (
+                <StyledCheckbox
+                  color="primary"
+                  disabled={hasAccess || hasRequestedAccess}
+                  id="countryIds"
+                  inputRef={register({ validate: (value: Country['id'][]) => value.length > 0 })}
+                  key={id}
+                  label={name}
+                  name="countryIds"
+                  tooltip={getTooltip()}
+                  value={id}
+                />
+              );
+            })}
+          </CountryChecklist>
+        </CountryChecklistWrapper>
+        <StyledBox>
+          <StyledFormInput
+            disabled={!project}
+            fullWidth
+            Input={TextField}
+            inputProps={{
+              enterKeyHint: 'done',
+              // Make <textarea> scroll upon overflow. MUI uses inline styling (element.style) to
+              // dynamically resize it to fit content, so we must do the same to override it.
+              style: sizeClassIsMdOrLarger ? { height: '100%', overflow: 'scroll' } : {},
```

Comment:
Can we do this just with CSS by targeting the class directly in a media query?

Reply:
Unfortunately not, because MUI applies height and overflow inline. As far as I know, that has *the* highest priority/specificity, and no CSS selector is able to override it 🥲

Reply:
I think we could make use of the `rows` prop on the text field. It should scroll automatically. There's a good example [here](https://v4.mui.com/components/text-fields/#multiline)

Reply:
This was a surprisingly gnarly problem. If we had a single-column layout, `rows` would be perfect. But because MUI puts the actual `<textarea>` in a wrapper (which has the border), this then becomes the element that determines the height of the entire section

I could:
1. hard-code the left and middle country lists to line up with a `FormInput` of height *n*, and leave an inline comment saying that their `height` values are coupled to the `rows` prop of a seemingly separate element
2. use inline styling to make it grow to fill the parent (current approach—hacky but semantically “correct")
3. just let the `<textarea>` be smaller than its container (looks kinda janky)

Reply:
I think we could give it a fixed `rows` value and make the other 2 columns fill the entire height to match the textbox. So let the textbox column dictate the height of the other 2 columns. Probably using `grid` or `flexbox` should make this possible

Reply:
This seems like a reasonable middle ground. I’ll be able to use this text box dictate the height of the middle column, but getting the list/table in the leftmost column to grow/shrink to a “cousin” element is apparently bit above my pay grade

Might retry this tomorrow with fresh eyes

Reply:
All good - I can help you if you need!